### DISCUSSION TITLE
docs/guide/project-setup

### DISCUSSION THREAD

**Discussion by giscus** [mikro-orm/mikro-orm#5225] [Comments]
# docs/guide/project-setup

So far we were just toying around with our entities, let's start building something real. We said we will use Fastify as a web server, and Vitest for testing it. Let's set that up, create our first endpoint and test it.

https://mikro-orm.io/docs/guide/project-setup

**Comment by kun-yu-tsai** [comment]
(Below my understanding might be wrong. Please see @B4nan 's comment)

in case someone struggle at here as me. There is a mis-comment in utils.ts:

```
export async function initTestApp(port: number) {
  // this will create all the ORM services and cache them
  const { orm } = await initORM({
    // no need for debug information, it would only pollute the logs
    debug: false,
    // we will use in-memory database, this way we can easily parallelize our tests
    dbName: ':memory:',
    // this will ensure the ORM discovers TS entities, with ts-node, ts-jest and vitest
    // it will be inferred automatically, but we are using vitest here
    tsNode: true,  // -> this line was originally comment out. We should leave it on.
  });
```

**Reply by B4nan** [reply]
this is not true, you don't need `tsNode: true` most of the time, it gets inferred automatically when you run tests with vitest or ts-jest

repro welcome, or just compare what you do differently with the actual project in https://github.com/mikro-orm/guide (which also does not have `tsNode: true` in there)

**Reply by kun-yu-tsai** [reply]
Thanks a lot! I was trying to set up the project and struggled a bit. Looks like I might misunderstood that as you commented. I'll look into the guide project and tried the full code. Appreciate it, mate.

**Comment by JShawnLooney** [comment]
Code snippet in the Initial migration section should be `npx mikro-orm-esm migration:create --initial` to utilize the ECMAScript module instead of `npx mikro-orm ...`

**Reply by B4nan** [reply]
Thanks for mentioning this, fixed via https://github.com/mikro-orm/mikro-orm/commit/d88cc91360c5e4bd283da97a41deddfffb33e4ef

I really need to find a way to unify this, maybe a bit of shell magic could help.

**Reply by JShawnLooney** [reply]
Btw, this tutorial is one of the better ones I've come across as I help our lead developer with ORM evaluations. It  is very helpful to see the ORM in context with Zod and an API.  Thanks for the great work.

**Comment by amerkarim** [comment]
Is there a reason you're using async for onClose but not for onRequest?

**Reply by B4nan** [reply]
`orm.close` is async, you need a promise to handle that, while the request context can be created with a callback instead. It's different kind of API.
